src/util/colorHelpers.js

```diff
@@ -68,35 +68,54 @@ export const calcNodeColor = (tree, colorScale) => {
 // scale entropy such that higher entropy maps to a grayer less-certain branch
 const branchInterpolateColour = "#BBB";
 const branchOpacityConstant = 0.6;
-export const branchOpacityFunction = scalePow()
+const branchOpacityFunction = scalePow()
   .exponent([0.6])
-  .domain([0, 2.0])
-  .range([0.4, 1])
+  .domain([0, 2.0]) // entropy values close to 0 -> ~100% confidence, close to 2 -> very little confidence
+  .range([0.4, 1])  // 0 -> return original node colour, 1 -> return branchInterpolateColour
   .clamp(true);
+const tipOpacityFunction = branchOpacityFunction
+  .copy()
+  .range([0, 0.9]); // if entropy close to 0 return the original node color
 
 
 // entropy calculation precomputed in augur
 // export const calcEntropyOfValues = (vals) =>
 //   vals.map((v) => v * Math.log(v + 1E-10)).reduce((a, b) => a + b, 0) * -1 / Math.log(vals.length);
 
 /**
- * calculate array of HEXs to actually be displayed.
- * (colorBy) confidences manifest as opacity ramps
+ * Calculate an array of stroke colors to render for a branch or tip node. These are "grey-er" versions
+ * of the underlying `tree.nodeColours`. The degree of grey-ness is obtained via interpolation
+ * between the node color and `branchOpacityConstant`. The interpolation parameter varies
+ * depending on the confidence we have in the trait (the entropy), with more confidence resulting
+ * in more saturated colours. For branches we always make them slightly greyer (even in the absence
+ * of uncertainty) for purely aesthetic reasons.
  * @param {obj} tree phyloTree object
+ * @param {bool} branch will this color be used for the branch or the tip?
  * @param {bool} confidence enabled?
  * @return {array} array of hex's. 1-1 with nodes.
  */
-export const calcBranchStrokeCols = (tree, confidence, colorBy) => {
+export const calculateStrokeColors = (tree, branch, confidence, colorBy) => {
```

Comment:
_not a request for change, just curious_

Why combine branch/tip colors into one function with the `branch` flag when they are essentially completely different paths within the function?

Reply:
While they are ~separate code paths, they both do the same thing: take a colour and modify it according to the node's uncertainty. Co-locating them feels natural to me and should help them to stay in-sync. 

I actually thought about taking this further and refactoring it into a function `calculateNodeColors -> {branchColors, tipStrokeColors, tipFillColors}` so we calculate everything at once, but I'll leave that for another day (and I need to check that we never have a situation where we only need to recompute one of those sets).